File: src/main/java/com/extendedae_plus/mixin/bridge/HelperCraftingJob.java
package com.extendedae_plus.mixin.bridge;

import appeng.api.crafting.IPatternDetails;
import appeng.api.stacks.GenericStack;
import appeng.crafting.CraftingLink;

import java.util.Map;

public interface HelperCraftingJob {
    Map<IPatternDetails, HelperJobProgress> getTasks();

    CraftingLink getLink();

    GenericStack getOutputFinal();

    Integer getPlayerID();

    long getRemainingAmount();

    interface HelperJobProgress {
        long getValue();
    }
}

File: src/main/kotlin/com/extendedae_plus/mixin/helper/HelperCraftingJob.kt
package com.extendedae_plus.mixin.helper

import appeng.api.crafting.IPatternDetails
import appeng.api.stacks.GenericStack
import appeng.crafting.CraftingLink

interface HelperCraftingJob {
    val tasks: MutableMap<IPatternDetails, HelperJobProgress>
    val link: CraftingLink
    val outputFinal: GenericStack
    val playerID: Int?
    val remainingAmount: Long

    interface HelperJobProgress {
        val value: Long
    }
}
